### PR TITLE
Objective C compiler plugin and runtime

### PR DESCRIPTION
**Pull request by thomasvl**
Alpha 1 drop of Google's Objective C support:
- protoc plugin/generator.
- Objective C Runtime Support.
- Xcode project for testing runtime (and generator).

### CODE REVIEW

**Comment by thomasvl** [comment]
That should fix the build failure for other platforms.

File: objectivec/GPBBootstrap.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#endif
+#endif
+
+// If property name starts with init we need to annotate it to get past ARC.
+// http://stackoverflow.com/questions/18723226/how-do-i-annotate-an-objective-c-property-with-an-objc-method-family/18723227#18723227
+#define GPB_METHOD_FAMILY_NONE __attribute__((objc_method_family(none)))
+
+// The protoc-gen-objc version which works with the current version of the
+// generated Objective C sources.
+#define GOOGLE_PROTOBUF_OBJC_GEN_VERSION 400
```
Comment (TeBoring) [inline objectivec/GPBBootstrap.h]: This line is also used for compiling generator. However, some syntax in this file cannot be used for gcc. (for example line 69). Could you move this line out of this file?

Comment (thomasvl) [inline objectivec/GPBBootstrap.h]: Yea, already working on a better fix.  Since I can't test it locally it is a little painful.

**Comment by thomasvl** [comment]
Dropping this, I'll setup Travis CI on my branch and get things passing and send out another pull request.

**Pull request closed without merging**